spec/syntax.md

```diff
@@ -704,3 +647,80 @@ s = 1*( SP / HTAB / CR / LF )
 The grammar is formally defined in [`message.abnf`](./message.abnf)
 using the ABNF notation,
 as specified by [RFC 5234](https://datatracker.ietf.org/doc/html/rfc5234).
+
+
+
+### Complex Messages
+
+The various features can be used to produce arbitrarily complex _messages_ by combining
+_declarations_, _selectors_, _functions_, and more.
+
+>A complex message with 2 _selectors_ and 3 local variable _declarations_:
+>
+>```
+>let $hostName = {$host :person firstName=long}
+>let $guestName = {$guest :person firstName=long}
+>let $guestsOther = {$guestCount :number offset=1}
+>
+>match {$host :gender} {$guestOther :number}
+>
+>when female 0 {{$hostName} does not give a party.}
+>when female 1 {{$hostName} invites {$guestName} to her party.}
+>when female 2 {{$hostName} invites {$guestName} and one other person to her party.}
+>when female * {{$hostName} invites {$guestName} and {$guestsOther} other people to her party.}
+>
+>when male 0 {{$hostName} does not give a party.}
+>when male 1 {{$hostName} invites {$guestName} to his party.}
+>when male 2 {{$hostName} invites {$guestName} and one other person to his party.}
+>when male * {{$hostName} invites {$guestName} and {$guestsOther} other people to his party.}
+>
+>when * 0 {{$hostName} does not give a party.}
+>when * 1 {{$hostName} invites {$guestName} to their party.}
+>when * 2 {{$hostName} invites {$guestName} and one other person to their party.}
+>when * * {{$hostName} invites {$guestName} and {$guestsOther} other people to their party.}
+>```
+
+
+> Expression examples:
```

Comment:
Did you have a place in mind for these expression examples, and/or a reason why not to include them in this doc?

Reply:
Along with "complex messages", I just pushed them out of the way for the nonce. I'm reluctant to delete them without looking more closely at the `expression` section. I'll refactor these into the text.